app/src/main/java/com/github/gotify/login/LoginActivity.java

```diff
@@ -115,11 +115,22 @@ public void doCheckUrl() {
         checkUrlProgress.setVisibility(View.VISIBLE);
         checkUrlButton.setVisibility(View.GONE);
 
-        final String fixedUrl = url.endsWith("/") ? url.substring(0, url.length() - 1) : url;
+        final String trimmedUrl = url.trim();
+        final String fixedUrl =
+                trimmedUrl.endsWith("/")
+                        ? trimmedUrl.substring(0, trimmedUrl.length() - 1)
+                        : trimmedUrl;
 
-        ClientFactory.versionApi(fixedUrl, tempSSLSettings())
-                .getVersion()
-                .enqueue(callInUI(this, onValidUrl(fixedUrl), onInvalidUrl(fixedUrl)));
+        try {
+            ClientFactory.versionApi(fixedUrl, tempSSLSettings())
+                    .getVersion()
+                    .enqueue(callInUI(this, onValidUrl(fixedUrl), onInvalidUrl(fixedUrl)));
+        } catch (Exception e) {
+            checkUrlProgress.setVisibility(View.GONE);
+            checkUrlButton.setVisibility(View.VISIBLE);
+            String errorMsg = getString(R.string.version_failed, url + "/version", e.getMessage());
```

Comment:
The fixedUrl should be used inside the error message, otherwise, there may be a double slash.
```suggestion
            String errorMsg = getString(R.string.version_failed, fixedUrl + "/version", e.getMessage());
```